python_modules/dagster/dagster/_core/definitions/asset_daemon_context.py

```diff
@@ -569,7 +569,11 @@ def build_run_requests(
                 )
             )
 
-    return run_requests
+    # We don't make public guarantees about sort order, but make an effort to provide a consistent
+    # ordering that puts earlier time partitions before later time partitions. Note that, with dates
+    # formatted like 12/7/2023, the run requests won't end up in time order. Sorting by converting
+    # to time windows would be more expensive, so we don't do it here.
+    return sorted(run_requests, key=lambda x: x.partition_key)
```

Comment:
is it really that expensive to call: https://sourcegraph.com/github.com/dagster-io/dagster/-/blob/python_modules/dagster/dagster/_core/definitions/asset_graph.py?L764:5-764:33? It doesn't end up doing the full time window / cron schedule process for basic hourly/daily partitions at least, and so should be fairly fast.

Fine with this approach if there are enough scenarios where this would be bad though, just seems a shame to have different sort orders throughout the product.

Reply:
I haven't had the chance to benchmark this and probably won't in the near future. What do you think about me merging this in its current state and potentially improving it later?

I'd update the comment to project less confidence that time windows is too expensive.

Reply:
Sure I think it's fine to merge as is